Sources/RestKit/Authentication/ConfigBasedAuthenticatorFactory.swift

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright IBM Corporation 2019
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+public enum AuthenticatorError: Error {
+    case noConfigurationFound
+    case authenticationTypeNotDefined
+    case authenticationTypeNotRecognized
+    case missingEnvironmentVariable(String)
+}
+
+public enum EnvironmentAuthenticatorType: String {
+    case IAM = "iam"
+    case Basic = "basic"
+    case CP4D = "cp4d"
+    case NoAuth = "noauth"
```

Comment:
I updated this file to make everything the proper case... in the case of the raw value "noauth", this is because our utility method that extracts env vars lowercases everything.